app/src/main/java/uk/gov/onelogin/ui/home/HomeScreenViewModel.kt

```diff
@@ -9,14 +9,16 @@ import kotlinx.coroutines.launch
 import uk.gov.android.authentication.TokenResponse
 import uk.gov.onelogin.repositiories.TokenRepository
 import uk.gov.onelogin.tokens.Keys
+import uk.gov.onelogin.tokens.usecases.GetEmail
 import uk.gov.onelogin.tokens.usecases.SaveToSecureStore
 import uk.gov.onelogin.tokens.usecases.SaveTokenExpiry
 
 @HiltViewModel
 class HomeScreenViewModel @Inject constructor(
     private val tokenRepository: TokenRepository,
     private val saveToSecureStore: SaveToSecureStore,
-    private val saveTokenExpiry: SaveTokenExpiry
+    private val saveTokenExpiry: SaveTokenExpiry,
+    val getEmail: GetEmail
```

Comment:
This should be private surely?

Then we should probably have some `LiveData` to handle the email in the ui

Reply:
as we discussed -> 1d72c76af5b623681ee7d7cd31d46f289a480372